app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java

```diff
@@ -56,23 +47,33 @@ public void writeToParcel(Parcel dest, int flags) {
     @Override
     public Where<Message, Object> modWhere(Where<Message, Object> where)
             throws SQLException {
-        where.like(Message.TIMESTAMP_FIELD, new SelectArg(date));
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        calendar.set(Calendar.HOUR_OF_DAY, 0);
+        calendar.set(Calendar.MINUTE, 0);
+        calendar.set(Calendar.SECOND, 0);
+        calendar.set(Calendar.MILLISECOND, 0);
+
+        Calendar calendar1 = Calendar.getInstance();
+        calendar1.setTime(calendar.getTime());
+        calendar1.add(Calendar.DAY_OF_MONTH, 1);
+        where.between(Message.TIMESTAMP_FIELD, calendar.getTime(), calendar1.getTime());
```

Comment:
ok so here we are creating 2 Calendar instances. I think we could use just one calendar instance
Example:
```
date1 = calendar.getTime();
calendar.add(DAY_OF_MONTH, 1);
date2 = calendar.getTime();
```